### DISCUSSION TITLE
"Ask copilot" input font issue

### DISCUSSION THREAD

**Discussion by Drahushchak** [community/community#64762] [Copilot Conversations] [answered]
### "Ask copilot" input font issue

Question

### Body

I have a problem with the "Ask copilot" functionality. Something has to do with the font. When I enter some prompt - the font of the input is strange, some of the characters are uppercase, some lowercase and some lowercase, but the size of an uppercase letter. The issue is on my windows PC. On mac - I use the same account with the settings synced, but have no issues. I tried resetting the font settings but it did not help

![image](https://github.com/community/community/assets/33318705/d1cc4f57-ca10-489a-9e0a-c410f2734b77)

**Comment by arafajs** [comment]
- Try completely uninstalling and reinstalling the Copilot extension to see if that resets the font settings.
- Check if you have any browser extensions installed that might be interfering with or overriding Copilot's font styling. Disable any suspicious extensions temporarily to test.
- Try using a different browser on your Windows PC to see if the issue persists. The font rendering may be specific to your current browser.
- Look in your Windows display settings/appearance settings to see if you have any system-wide font options enabled that could override application fonts.
- Update your graphics drivers and browser to the latest versions in case this is a rendering glitch.

**Reply by Drahushchak** [reply]
- Tried to reinstall extension did not help.
- I don't see trying the browser stuff, as I am using the vscode app, not from the browser.
- Graphics card drivers are updated

**Reply by arafajs** [reply]
1. Update Visual Studio Code to the latest version. There could be a bug with Copilot's font rendering in older versions that is fixed in a newer release.
2. Try resetting VS Code to default settings by going to File > Preferences > Settings and clicking the "Reset All Settings" button. This will revert any custom font configs.
3. Check if you have any other VS Code extensions enabled that might conflict with Copilot. Disable other extensions temporarily to see if that helps.
4. Try changing the VS Code theme to something simple like Light or Dark to eliminate any theme-specific font styling interference.
5. Open the Copilot font settings (Code > Preferences > Settings > Extensions > Copilot Configuration) and play around with the font family, size, etc to see if you can force a change.
6. As a workaround, use the VS Code command Palette > Toggle Copilot Markdown Rendering to make Copilot output plain text without styling.

**Reply by Drahushchak** [reply]
Resetting VS Code to default settings helped. Now I need to figure out what exact settings were causing that issue. Once I find the answer - I will post it here.

**Reply by Drahushchak** [reply, marked as answer]
The issue was the font ligatures that I was using.
```json
"editor.fontLigatures": "'ss01', 'ss02', 'ss03', 'ss04', 'ss05', 'ss06', 'ss07', 'ss08', 'zero', 'onum'",
```
I have set it to
```json
"editor.fontLigatures": true,
```
But the problem is that some of the letters don't look the same as they used to, but it resolved the issue.
For reference: I am using the `Cascadia Code PL` font.

**UPDATED**:
changing the setting (removing `'ss02'`) to:
```json
"editor.fontLigatures": "'ss01', 'ss03', 'ss04', 'ss05', 'ss06', 'ss07', 'ss08', 'zero', 'onum'",
```
also helped (some characters are still different, but not as bad as it was).
